lib/internal/process/promises.js

```diff
@@ -29,11 +29,28 @@ function handler(type, promise, reason) {
   }
 }
 
+// Feature detection for easier backporting. Even though it's semver-major it
+// is safe to backport this and therefore making backports in general easier.
+let warnedSwallowed = Number(
+  process.version.slice(1, process.version.indexOf('.'))
+) < 11;
+
 function resolveError(type, promise, reason) {
   // We have to wrap this in a next tick. Otherwise the error could be caught by
   // the executed promise.
   process.nextTick(() => {
-    process.emit('multipleResolves', type, promise, reason);
+    if (!process.emit('multipleResolves', type, promise, reason) &&
+        !warnedSwallowed) {
+      // eslint-disable-next-line no-restricted-syntax
+      const warning = new Error(
+        'A promise constructor called `resolve()` or `reject()` more than ' +
+        'once or both function together. It is recommended to use ' +
```

Comment:
typo: functions

(or just omit “or both functions together”, it sounds a bit to me like they’d be called at the same time?)